core/templates/pages/story-editor-page/navbar/story-editor-navbar.component.spec.ts

```diff
@@ -106,20 +125,393 @@ describe('Story editor navbar directive', () => {
+      spyOn(storyEditorStateService, 'getStoryWithUrlFragmentExists')
+        .and.returnValue(true);
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+
+      mockStoryReinitializedEventEmitter.emit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues).toContain(
+        'Story URL fragment already exists.');
+    });
+
+    it('should show validation error when chapters in story ' +
+      'does not have any linked exploration', fakeAsync(() => {
+      // Setting exploration ID to be empty.
+      storyBackendDict.story_contents.nodes[0].exploration_id = null;
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockStoryReinitializedEventEmitter = new EventEmitter();
+
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+      spyOnProperty(storyEditorStateService, 'onStoryInitialized')
+        .and.returnValue(mockStoryReinitializedEventEmitter);
+      spyOn(storyEditorStateService, 'getStoryWithUrlFragmentExists')
+        .and.returnValue(true);
+      spyOn(editableStoryBackendApiService, 'validateExplorationsAsync')
+        .and.returnValue(Promise.resolve([]));
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.explorationValidationIssues.length).toBe(0);
+
+      mockStoryReinitializedEventEmitter.emit();
+      tick();
+      fixture.detectChanges();
+
+      expect(component.explorationValidationIssues).toContain(
+        'Some chapters don\'t have exploration IDs provided.');
+    }));
+
+    it('should validate story without any validation errors ' +
+      'on initialization', () => {
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockStoryInitializedEventEmitter = new EventEmitter();
+
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+      spyOnProperty(storyEditorStateService, 'onStoryInitialized')
+        .and.returnValue(mockStoryInitializedEventEmitter);
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+
+      mockStoryInitializedEventEmitter.emit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+    });
+
+    it('should validate story without any validation errors ' +
+      'on reinitalization', () => {
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockStoryReinitializedEventEmitter = new EventEmitter();
+
+      spyOnProperty(storyEditorStateService, 'onStoryReinitialized')
+        .and.returnValue(mockStoryReinitializedEventEmitter);
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+
+      mockStoryReinitializedEventEmitter.emit();
```

Comment:
I think it won't be necessary since initializing, reinitializing, and undo method uses EventEmitter to trigger changes, we are mocking event emitter in the test file.
If we were to add tests it would be simply changing the variable name.